pyrft/other.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+"""
+ARI semi modified function files
+"""
+import numpy as np
+from scipy.ndimage import label
+from scipy.stats import norm
+
+from nilearn.input_data import NiftiMasker
+from nilearn.image import get_data, math_img
+
+def ari(pvalues, subset, alpha= 0.1 ):
+    """ ari implements all resolution inference (Rosenblatt 2018, see below) and 
+    calulates bounds on the TDP for p-values within the specified subset
+
+    Parameters
+    ----------
+    pvalues:  a numpy.nd array,
+        containing the p-values
+
+    subset:   numpy boolean array,
+        the same size as pvalues indicating which elements
+        to use in the subset or 
+
+    alpha: float or list, optional
+        level of control on the true positive rate, aka true dsicovery
+        proportion
+
+    Returns
+    -------
+    tdp:  float,
+        the true discovery proportion found by applying ARI
+
+    Note
+    ----
+    This implements the method described in:
+
+    Rosenblatt JD, Finos L, Weeda WD, Solari A, Goeman JJ. All-Resolutions
+    Inference for brain imaging. Neuroimage. 2018 Nov 1;181:786-796. doi:
+    10.1016/j.neuroimage.2018.07.060
+    """
+
+    hommel_value = _compute_hommel_value(pvalues, alpha)
+    subset_pvalues = pvalues[subset]
+    tdp = _true_positive_fraction(subset_pvalues, hommel_value, alpha)
+    
+    return tdp
+
+def _compute_hommel_value(pvalues, alpha = 0.1):
+    """ This function computes the hommel value needed for all resolution 
+    inference
+
+    Parameters
+    ---------------------
+    pvalues:      a numpy.nd array,
+        containing the pvalues
+    alpha:        float,    
+        the significance level, default is 0.1
+
+    Returns
+    ---------------------
+    hommel_value: the value from implementing the hommel procedure
+
+    Examples
+    ---------------------
+zvals = np.random.randn(1, 100)

```

Comment:
Still an indentation problem here

Reply:
As mentioned in my comments to your previous comments - this is a deliberate choice. Did you get a chance to look at my responses?

Reply:
But you cannot break conventions for personal convenience.

There is no value in having negotiations about coding style: these are community conventions that neither you nor me are responsible for. I don't like some of them either, but this is not the point.
I'm sorry to bother you with that, but these things should be taken in charge by automatic tools, eg black.
http://www.locallyoptimal.com/blog/2019/08/23/why-you-should-use-black-for-your-python-style-linting/
Maybe we should rely on that.